rest-service/manager_rest/rest/resources_v3/secrets.py

```diff
@@ -132,28 +132,36 @@ def _encrypt_secret_value(self, value, encryption_key=None):
                          config.instance.security_encryption_key
         return cryptography_utils.encrypt(encryption_key, value)
 
-    def _is_value_permitted(self, creator):
+    def _is_hidden_value_permitted(self, creator):
         current_tenant = get_storage_manager().current_tenant
         current_username = current_user.username
         return is_administrator(current_tenant) or creator == current_username
 
     def _validate_secret_modification_permitted(self, secret):
         get_resource_manager().validate_modification_permitted(secret)
         if secret.is_hidden_value and \
-                not self._is_value_permitted(secret.created_by):
+                not self._is_hidden_value_permitted(secret.created_by):
             raise ForbiddenError(
                 'User `{0}` is not permitted to modify the hidden value '
                 'secret `{1}`'.format(current_user.username, secret.key)
             )
 
     def _update_is_hidden_value(self, secret):
         is_hidden_value = request.json.get('is_hidden_value')
-        if is_hidden_value is not None:
-            is_hidden_value = rest_utils.verify_and_convert_bool(
-                'is_hidden_value',
-                is_hidden_value
+        if is_hidden_value is None:
+            return
+        is_hidden_value = rest_utils.verify_and_convert_bool(
```

Comment:
can do this after _is_hidden_value_permitted

Reply:
I think it is better before because it also validate that is_hidden_value is a valid boolean